### PR TITLE
Fix #4448 Open context menu of datatable after update

### LINKED ISSUE #4448: DataTable/TreeTable: ContextMenu disappears after update
I have a dataTable with contextMenu and menuItem dynamically being shown/disabled or not in it.

More or less it looks like this:

`<p:dataTable id="dataTableId">
.....
<p:ajax event="contextMenu" 
listener="#{controller.onMenuDisplay} 
update="contextMenuId" />
.....
</p:dataTable>
`

And below:

`
<p:contextMenu id="contextMenuId" for="dataTableId"
widgetVar="contextMenuWidgetVar">
					<p:menuitem value="XXX"
								process="@this"
								actionListener="#{controller.action}"
								rendered="#{controller.shouldBeRendered}"/>
`
After executing ajax event, contextMenu shows for a while and disappers.

1) Environment
- PrimeFaces version: 6.2.14
- Does it work on the newest released PrimeFaces version? Version? No
- Does it work on the newest sources in GitHub? (Build by source -> https://github.com/primefaces/primefaces/wiki/Building-From-Source) No
- Application server + version: WLS 12.2.3.0.0
- Affected browsers: All (FireFox, Chrome, Edge, IE)

2) Expected behavior

Context menu must update its menuItems and stay visible

3) Actual behavior

Context menu updates, shows itself for a while and disappears.

4) Steps to reproduce
Click with RMB on a row in a dataTable with p:ajax event="contextMenu" with update being set on contextMenu with rendered and disabled parameters.

### PR DESCRIPTION
**Pull request by naofireblade**
Applies the fix from #5069 to datatable.

Resolves #4448

### CODE REVIEW

**Comment by naofireblade** [comment]
Works for me but probably needs some more testing with all datatable options and events.

**Comment by melloware** [comment]
Looks like its failing our JSDOC check...
```
[INFO] Error: [tsDiagnosticsError] at <unknown> (/home/runner/work/primefaces/primefaces/target/type-declaration/PrimeFaces.d.ts:9024:87)
[INFO] Cannot find name 'function'.
[INFO]     at handleAndGetError (/home/runner/work/primefaces/primefaces/src/main/type-definitions/src/error.js:115:21)
[INFO]     at /home/runner/work/primefaces/primefaces/src/main/type-definitions/src/ts-postprocess-defaulthooks.js:87:57
[INFO]     at Array.flatMap (<anonymous>)
[INFO]     at /home/runner/work/primefaces/primefaces/src/main/type-definitions/src/ts-postprocess-defaulthooks.js:87:44
[INFO]     at invokeHook (/home/runner/work/primefaces/primefaces/src/main/type-definitions/src/ts-postprocess.js:19:24)
[INFO]     at /home/runner/work/primefaces/primefaces/src/main/type-definitions/src/ts-postprocess.js:92:17
[INFO]     at asyncFlatMap (/home/runner/work/primefaces/primefaces/src/main/type-definitions/src/lang.js:74:33)
[INFO]     at validateProgram (/home/runner/work/primefaces/primefaces/src/main/type-definitions/src/ts-postprocess.js:90:26)
[INFO]     at async postprocessDeclarationFile (/home/runner/work/primefaces/primefaces/src/main/type-definitions/src/ts-postprocess.js:203:19)
[INFO]     at async main (/home/runner/work/primefaces/primefaces/src/main/type-definitions/src/cli.js:591:30)
```

**Comment by melloware** [comment]
@blutorange can you check this? @naofireblade has added a function callback and his syntax for the JSDOC is wrong.

File: src/main/resources/META-INF/resources/primefaces/datatable/datatable.js [hunk 1]
```diff
@@ -2396,8 +2402,9 @@ PrimeFaces.widget.DataTable = PrimeFaces.widget.DeferredWidget.extend({
      * @param {JQuery.TriggeredEvent} event Event that occurred.
      * @param {JQuery} rowElement Row that was clicked.
      * @param {PrimeFaces.widget.DataTable.CmSelectionMode} cmSelMode The current selection mode.
+     * @param {function} fnShowMenu callback to open menu.
```
Comment (blutorange) [inline src/main/resources/META-INF/resources/primefaces/datatable/datatable.js]: @naofireblade You need to specify the signature of the callback function (as a TypeScript type). In this case, from what I can see, I would guess it should be

```
     * @param {() => void} [fnShowMenu] Optional callback function invoked when the menu was opened.
``` 

The `[...]` indicate that the parameter is optional.

Same goes for the `fireRowSelectEvent` method

Comment (naofireblade) [reply src/main/resources/META-INF/resources/primefaces/datatable/datatable.js]: I'm new to ts and had no idea how to classify an optional callback function in the parameter documentation.

Comment (blutorange) [reply src/main/resources/META-INF/resources/primefaces/datatable/datatable.js]: Yeah no problem we're not blaming you : )

Comment (naofireblade) [reply src/main/resources/META-INF/resources/primefaces/datatable/datatable.js]: Thanks for your help! Will fix it later today.

**Comment by melloware** [comment]
Looks like the build is passing now.  Nice teamwork.

**Comment by melloware** [comment]
I think its safe to take this PR out of draft mode.

**Review by melloware — APPROVED** [review]

**Comment by melloware** [comment]
@naofireblade any interest in adding this same code to TreeTable as well?

**Comment by naofireblade** [comment]
Yes I should have time for this next week.

**Comment by melloware** [comment]
@naofireblade I merged this one so you can open a new PR for TreeTable.  Thanks for your contribution!

**Comment by melloware** [comment]
@naofireblade I am running the Showcase Datatable ContextMenu example locally and its no longer working.

**Comment by naofireblade** [comment]
> @naofireblade I am running the Showcase Datatable ContextMenu example locally and its no longer working.

Thanks, I will check that.

**Comment by melloware** [comment]
I fixed it. You were missing the Non-AJAX scenario.  See PR: https://github.com/primefaces/primefaces/pull/7376

**Pull request merged**